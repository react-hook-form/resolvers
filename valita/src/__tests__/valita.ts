/* eslint-disable no-console, @typescript-eslint/ban-ts-comment */
import { valitaResolver } from '..';
import {
  fields,
  invalidData,
  invalidSchemaErrorData,
  schema,
  schemaError,
  validData,
  validSchemaErrorData,
} from './__fixtures__/data';

const shouldUseNativeValidation = false;
describe('valitaResolver', () => {
  it('should return values from valitaResolver when validation pass', async () => {
    const result = await valitaResolver(schema)(validData, undefined, {
      fields,
      shouldUseNativeValidation,
    });

    expect(result).toEqual({ errors: {}, values: validData });
  });

  it('should return a single error from valitaResolver when validation fails', async () => {
    const result = await valitaResolver(schema)(invalidData, undefined, {
      fields,
      shouldUseNativeValidation,
    });

    expect(result).toMatchSnapshot();
  });

  it('should return all the errors from valitaResolver when validation fails with `validateAllFieldCriteria` set to true', async () => {
    const result = await valitaResolver(schema)(invalidData, undefined, {
      fields,
      criteriaMode: 'all',
      shouldUseNativeValidation,
    });

    expect(result).toMatchSnapshot();
  });

  it('should be able to validate variants without errors', async () => {
    const result = await valitaResolver(schemaError)(
      validSchemaErrorData,
      undefined,
      {
        fields,
        shouldUseNativeValidation,
      },
    );

    expect(result).toEqual({
      errors: {},
      values: {
        type: 'a',
      },
    });
  });

  it('should be able to validate invalid schema with errors', async () => {
    const result = await valitaResolver(schemaError)(
      invalidSchemaErrorData,
      undefined,
      {
        fields,
        shouldUseNativeValidation,
      },
    );

    expect(result).toEqual({
      errors: {
        type: {
          message: 'invalid_literal',
          ref: undefined,
          type: 'invalid_literal',
        },
      },
      values: {},
    });
  });
});
