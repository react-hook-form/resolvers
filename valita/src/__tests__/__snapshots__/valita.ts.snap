// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`valitaResolver > should return a single error from valitaResolver when validation fails 1`] = `
{
  "errors": {
    "accessToken": {
      "message": "missing_value",
      "ref": undefined,
      "type": "missing_value",
    },
    "birthYear": {
      "message": "invalid_type",
      "ref": undefined,
      "type": "invalid_type",
    },
    "email": {
      "message": "Invalid email address",
      "ref": {
        "name": "email",
      },
      "type": "custom_error",
    },
    "enabled": {
      "message": "missing_value",
      "ref": undefined,
      "type": "missing_value",
    },
    "like": {
      "id": {
        "message": "invalid_type",
        "ref": undefined,
        "type": "invalid_type",
      },
      "name": {
        "message": "missing_value",
        "ref": undefined,
        "type": "missing_value",
      },
    },
    "password": {
      "message": "One uppercase character",
      "ref": {
        "name": "password",
      },
      "type": "custom_error",
    },
    "repeatPassword": {
      "message": "missing_value",
      "ref": undefined,
      "type": "missing_value",
    },
    "tags": [
      {
        "message": "invalid_type",
        "ref": undefined,
        "type": "invalid_type",
      },
      {
        "message": "invalid_type",
        "ref": undefined,
        "type": "invalid_type",
      },
      {
        "message": "invalid_type",
        "ref": undefined,
        "type": "invalid_type",
      },
    ],
    "username": {
      "message": "missing_value",
      "ref": {
        "name": "username",
      },
      "type": "missing_value",
    },
  },
  "values": {},
}
`;

exports[`valitaResolver > should return all the errors from valitaResolver when validation fails with \`validateAllFieldCriteria\` set to true 1`] = `
{
  "errors": {
    "accessToken": {
      "message": "missing_value",
      "ref": undefined,
      "type": "missing_value",
    },
    "birthYear": {
      "message": "invalid_type",
      "ref": undefined,
      "type": "invalid_type",
    },
    "email": {
      "message": "Invalid email address",
      "ref": {
        "name": "email",
      },
      "type": "custom_error",
    },
    "enabled": {
      "message": "missing_value",
      "ref": undefined,
      "type": "missing_value",
    },
    "like": {
      "id": {
        "message": "invalid_type",
        "ref": undefined,
        "type": "invalid_type",
      },
      "name": {
        "message": "missing_value",
        "ref": undefined,
        "type": "missing_value",
      },
    },
    "password": {
      "message": "One uppercase character",
      "ref": {
        "name": "password",
      },
      "type": "custom_error",
    },
    "repeatPassword": {
      "message": "missing_value",
      "ref": undefined,
      "type": "missing_value",
    },
    "tags": [
      {
        "message": "invalid_type",
        "ref": undefined,
        "type": "invalid_type",
      },
      {
        "message": "invalid_type",
        "ref": undefined,
        "type": "invalid_type",
      },
      {
        "message": "invalid_type",
        "ref": undefined,
        "type": "invalid_type",
      },
    ],
    "username": {
      "message": "missing_value",
      "ref": {
        "name": "username",
      },
      "type": "missing_value",
    },
  },
  "values": {},
}
`;
